core/src/main/scala/org/http4s/Message.scala

```diff
@@ -20,6 +22,16 @@ sealed trait Message extends MessageOps {
   def headers: Headers
   
   def body: EntityBody
+
+  final def bodyAsText(implicit defaultCharset: Charset = DefaultCharset): Process[Task, String] = {
+    (charset getOrElse defaultCharset) match {
+      case Charset.`UTF-8` => body |> utf8Decode
+      case cs =>
+        // TODO broken if multi-byte sequences don't break cleanly on chunk boundaries
+        body map { chunk => new String(chunk.toArray, cs.nioCharset) }
```

Comment:
Yeah, this is a bummer. A general `textDecoder` would be good to have in streams proper.